conn2res/task.py

```diff
@@ -99,21 +142,40 @@ def classification(x, y, **kwargs):
     return score
 
 
-def multiClassClassification(x, y, **kwargs):
+def multiClassClassification(x, y, model=None, **kwargs):
     """
     Multi-class Classification tasks
-    #TODO
+    # TODO
     """
 
+    # pop variables from kwargs
+    sample_weight_train, sample_weight_test = kwargs.pop(
+        'sample_weight', (None, None))
+
     x_train, x_test = x
     y_train, y_test = y
 
-    # capture only decision time points
-    idx_train = np.nonzero(y_train)
-    idx_test  = np.nonzero(y_test)
+    # specify default model
+    if model is None:
+        model = OneVsRestClassifier(RidgeClassifier(
+            alpha=0.0, fit_intercept=False, **kwargs))
 
-    model = OneVsRestClassifier(RidgeClassifier(alpha=0.0, fit_intercept=False, **kwargs)).fit(x_train[idx_train], y_train[idx_train])
-    score = model.score(x_test[idx_test], y_test[idx_test])
+    if isinstance(model, OneVsRestClassifier):
+        # select decision time points
+        idx_train = np.nonzero(y_train)
+        idx_test = np.nonzero(y_test)
+
+        # fit model on training data (OneVsRestClassifier does not support sample weights)
+        model.fit(x_train[idx_train], y_train[idx_train])
+
+        # calculate model scores on test data
+        score = model.score(x_test[idx_test], y_test[idx_test])
```

Comment:
I kept the usage of idx_train and idx_test for backward compatibility. However, as we discussed in our meeting, it makes more sense to train on all data and test as decision time points.

With the new sample_weight_test functionality, the decision time points can be defined there as a Boolean, i.e. passing weight=1 for at decision time points and weight=0 elsewhere.